xcdat/bounds.py

```diff
@@ -560,91 +573,86 @@ def get_daily_time_bounds(time, frequency=1):
     Raises
     ------
     ValueError
-        If an incorrect ``frequency`` argument is passed. Should be
-        2, 3, 4, 6, 8, 12, or 24.
+        If an incorrect ``freq`` argument is passed. Should be 1, 2, 3, 4, 6, 8,
+        12, or 24.
 
     Notes
     -----
-    This function is intended to reproduce CDAT's setAxisTimeBoundsDaily
-    method [1]_.
+    This function is intended to reproduce CDAT's ``setAxisTimeBoundsDaily``
+    method [3]_.
 
     References
     ----------
-    [1] https://github.com/CDAT/cdutil/blob/master/cdutil/times.py#L1093
+    .. [3] https://github.com/CDAT/cdutil/blob/master/cdutil/times.py#L1093
     """
-    # get calendar
-    calendar = time.encoding["calendar"]
+    if (freq > 24) | (np.mod(24, freq)):
+        raise ValueError(
+            "Incorrect `freq` argument. Supported values include 1, 2, 3, 4, 6, 8, 12, "
+            "and 24."
+        )
 
-    # get cftime class to create new cftime objects
+    # Get the calendar type from the time coordinates to determine the `cftime`
+    # object to represent bounds values.
+    calendar = time.encoding["calendar"]
     cf_obj = get_date_type(calendar)
 
-    # loop over time values and compute bounds
+    # Loop over the time values to generate a bounds DataArray.
     time_bnds = []
-    if (frequency > 24) | (np.mod(24, frequency)):
-        raise ValueError(
-            "Incorrect `frequency` argument."
-            " Supported values include 2, 3, "
-            "4, 6, 8, 12, and 24."
-        )
     for step in time.values:
-        # get year / month
         y, m, d, h = step.year, step.month, step.day, step.hour
-        for f in range(frequency):
-            if f * (24 // frequency) <= h < (f + 1) * (24 // frequency):
-                l_bnd = cf_obj(y, m, d, f * (24 // frequency))
-                u_bnd = l_bnd + datetime.timedelta(hours=(24 // frequency))
-        # store
+        for f in range(freq):
+            if f * (24 // freq) <= h < (f + 1) * (24 // freq):
+                l_bnd = cf_obj(y, m, d, f * (24 // freq))
+                u_bnd = l_bnd + datetime.timedelta(hours=(24 // freq))
         time_bnds.append([l_bnd, u_bnd])
 
-    # create dataarray
-    time_bnds = xr.DataArray(  # type: ignore
+    da_time_bnds = xr.DataArray(
         name=f"{time.name}_bnds",
         data=time_bnds,
         coords=dict(time=time),
         dims=[*time.dims, "bnds"],
         attrs={"xcdat_bounds": "True"},
     )
 
-    return time_bnds
+    return da_time_bnds
 
 
-def get_time_bounds(time):
-    """Sets the time bounds for a time coordinate axis.
+def create_time_bounds(time: xr.DataArray) -> xr.DataArray:
```

Comment:
This higher-level function should ordered based on the call stack and ahead of its children functions (`create_X_time_bounds()`).

Reply:
Note that I didn't add to this PR again – so this is still a to do item.